pkg/user/password_rest.go

```diff
@@ -0,0 +1,38 @@
+package user
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/oklog/ulid"
+)
+
+type PasswordReset struct {
+	Token     string
+	CreatedAt time.Time
+	ExpiresAt time.Time
+	IsUsed    bool
+}
+
+func NewPasswordReset() *PasswordReset {
+	return &PasswordReset{
+		Token:     generateToken(),
+		CreatedAt: time.Now(),
+		ExpiresAt: time.Now().Add(24 * time.Hour),
+		IsUsed:    false,
+	}
+}
+
+func generateToken() string {
+	t := time.Now()
+	entropy := ulid.Monotonic(rand.New(rand.NewSource(t.UnixNano())), 0)
```

Comment:
I think UUIDv4 is better because advantages of ULID don't mean much this usecase.

ULID adbantages:
- Short
- URL safe -> No sense for this usecase
- Timestamp is embed and sortable -> No sense for this usecase

ULID weakness:
- Short so easy to guess -> I'm somewhat nervous about using ULID for this use case.
- Timestamp ca be read by anyone
- Clocks must be synchronized to be used in distributed systems